crucible-mir/src/Mir/Trans.hs

```diff
@@ -1438,6 +1438,17 @@ union's fields. When interpreting this initialization:
 - We declare that the given field spans the entire `MirAggregate` representing
   the union, even if the field type's size on its own would be smaller than the
   size of the union/`MirAggregate`.
+
+The type of a (subrange of a) `MirAggregate` is unspecified until it's written
+to, and fixed thereafter. This means that, once a union's `MirAggregate` is
+initialized with a field of a given type, we only support reading from the union
+via a field of the same type (which, in practice, generally means the same
+field). When reading from the union, we rely on the initialization behavior
+described above, by reading the entire `MirAggregate` starting from offset 0.
+
+To properly implement reinterpretation of union values at other types, we'd need
+to change the behavior of `MirAggregate` to support type-switching, and we'd
+need to mimic Rust's layout rules for unions.
```

Comment:
After opening an issue about supported type-reinterpreting unions, it would be worth citing that issue here.